AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -147,7 +148,18 @@ - (void)_calculateSize
   if (isinf(_constrainedSize.width) == NO && [_attributes.pointSizeScaleFactors count] > 0) {
     _currentScaleFactor = [[self fontSizeAdjuster] scaleFactor];
   }
-  
+
+  // If we do not scale, do exclusion, or do custom truncation, we should just use TextKit for a fast-path.
+  BOOL doesNotScale = _currentScaleFactor == 1;
+  // NOTE: This code does not correctly handle if they set `…` with different attributes.
+  BOOL defaultTruncation = _attributes.avoidTailTruncationSet == nil && [_attributes.truncationAttributedString.string isEqualToString:@"\u2026"];
+  BOOL doesNoExclusion = _attributes.exclusionPaths.count == 0;
+  if (doesNotScale && defaultTruncation && doesNoExclusion) {
+    CGSize baseSize = [_attributes.attributedString boundingRectWithSize:_constrainedSize options:NSStringDrawingUsesLineFragmentOrigin | NSStringDrawingTruncatesLastVisibleLine context:nil].size;
```

Comment:
We should try running Pinterest in the CJK / scaled languages (Japanese is a good one for testing).  I think Garrett is familiar with what to look for, if Ricky isn't back on Monday.